patch/bar_tags.c

```diff
@@ -49,11 +49,23 @@ draw_tags(Bar *bar, BarArg *a)
 		invert = 0;
 		w = TEXTW(icon);
 		drw_setscheme(drw, scheme[
+			#if BAR_COLORFULTAGS_PATCH
+			#if BAR_UNDERLINETAGS_PATCH
+			(occ | m->tagset[m->seltags]) & 1 << i
+			#else
+			m->tagset[m->seltags] & 1 << i
+			#endif // BAR_UNDERLINETAGS_PATCH
+			? m->colorseltag
+				? tagschemes[i]
```

Comment:
This is fine as is, but I just wanted to note that if the user configures their build to have, say, 11 tags then dwm will likely crash at some point referring to values that are out of bounds here.

One way to avoid this could be to do something like `tagschemes[i % NUMTAGS]` at least then it would wrap around.

On the same note the entire tagschemes array is redundant in this context as you could just replace the above with

```c
? SchemeTag1 + i
```

(of course you still risk running out of color schemes if increasing the number of tags)

Reply:
You're right about `tagschemes`, but if we remove that, we'll have to assume that the enums are declared sequentially. Would it be safe to assume that?
Regarding the number of tags, I guess we could mention in the patch description that the user needs to define more colors if they are using more than 9 tags.